### ISSUE TITLE
MD051 silently skips fragments in repository-absolute cross-file links

### ISSUE THREAD

**Issue by cbachhuber** [rvben/rumdl#862]
### rumdl version

0.2.69

### Installation method

Pre-built binary (via `uvx rumdl@0.2.69`, PyPI wheel)

### Bug description

`MD051` validates a fragment in a relative cross-file link (`x.md#heading`) but not in the absolute form of the same link (`/docs/x.md#heading`).
The absolute link is not reported as unresolvable either — it is simply never checked, so a broken section link passes silently.

This matters for repositories that deliberately use repository-absolute links.
GitHub renders `/docs/x.md` from the repository root, and `MD057` already supports exactly this via `absolute-links = "relative_to_roots"`, so it is surprising that `MD051` does not follow.

Unlike `MD057`, `MD051` also offers no way to opt in.
Adding the same keys to the `[MD051]` section is rejected:

```
[config warning] Unknown option for rule MD051: absolute-links
[config warning] Unknown option for rule MD051: roots
```

### Steps to reproduce

```
/tmp/repro/
├── .rumdl.toml
├── adir/some.md
└── docs/
    ├── t.md
    └── x.md
```

`.rumdl.toml`:

```toml
[MD057]
absolute-links = "relative_to_roots"
roots = ["."]
```

`docs/t.md`:

```markdown
# T

[rel cross-file frag](x.md#missing-heading)

[abs cross-file frag](/docs/x.md#missing-heading)
```

`docs/x.md`:

```markdown
# X

## Other Heading
```

Run `rumdl check .` from `/tmp/repro`.

### Expected behavior

Both links are reported, because neither `#missing-heading` exists in `docs/x.md`:

```
docs/t.md:3:1: [MD051] Link fragment 'missing-heading' not found in 'x.md'
docs/t.md:5:1: [MD051] Link fragment 'missing-heading' not found in '/docs/x.md'
```

### Actual behavior

Only the relative link is reported:

```
docs/t.md:3:1: [MD051] Link fragment 'missing-heading' not found in 'x.md'

Issues: Found 1 issues in 1/3 files (11ms)
```

### Configuration

See the `.rumdl.toml` above.

### Operating system

Linux

### Additional context

Reading through `main`, the cause looks like path resolution rather than an intentional exclusion:

- `MD051::is_cross_file_link` in `src/rules/md051_link_fragments.rs` explicitly classifies a `/`-prefixed destination as a cross-file link ("These are links to other pages on the same site") and returns `true`, so the link is indexed rather than discarded.
- `link_target_candidates` in `src/workspace_index.rs` then resolves it with `source_file.parent().join(target_path)`. Rust's `Path::join` discards the base when the argument is absolute, so the candidate becomes the filesystem-absolute `/docs/x.md` rather than the workspace-relative `docs/x.md`.
- `cross_file_check` looks that candidate up in the workspace index, misses, and falls through its final branch: `// If target file not in index, skip (could be external file or not in workspace)`.

That last branch is what makes the failure silent — an absolute link is indistinguishable from a genuinely out-of-workspace target.

Two possible directions, depending on what you consider in scope:

1. Resolve `/`-prefixed targets against the project root (and any `MD057` `roots`), so `MD051` matches `MD057`'s existing behavior. This would mean sharing the `absolute-links`/`roots` configuration between the two rules, or having `MD051` read `MD057`'s.
2. If absolute links are intentionally out of scope for `MD051`, it would help to say so in `docs/md051.md`. It currently states that cross-file fragment links are validated without qualifying that to relative paths, which is what led me here.

Happy to open a PR for either direction if you have a preference.

**Comment by cbachhuber** [comment]
Thank you for the quick turnaround! Same remark as in https://github.com/rvben/rumdl/issues/863#issuecomment-5679933785
